src/Storages/IKVStorage.h

```diff
@@ -0,0 +1,28 @@
+#pragma once
+
+#include <Storages/IStorage.h>
+
+namespace DB
+{
+
+
+/// Storage that support key-value requests
+class IKeyValueStorage : public IStorage
```

Comment:
Another option is to add these methods to IStorage with default implementations throwing exceptions.